quil-rs/src/program/memory.rs

```diff
@@ -315,12 +315,9 @@ impl Expression {
 impl WaveformInvocation {
     /// Return, if any, the memory references contained within this WaveformInvocation.
     pub fn get_memory_references(&self) -> Vec<&MemoryReference> {
-        let mut result = vec![];
-
-        for expression in self.parameters.values() {
-            result.extend(expression.get_memory_references());
-        }
-
-        result
+        self.parameters
+            .values()
+            .flat_map(Expression::get_memory_references)
+            .collect()
```

Comment:
Made this change to see if it made any performance difference, left it because it didn't hurt anything and is more readable, IMO